src/DurableTask.AzureStorage/Partitioning/TablePartitionManager.cs

```diff
@@ -0,0 +1,788 @@
+                                    releasedLease = true;
+                                    await task;// catch exception for drain tasks
+                                }
+                            }
+                            else // If lease hasn't start draining, then drain it.
+                            {
+                                this.DrainLease(partition, CloseReason.LeaseLost);
+                                drainedLease = true;
+                            }
+                        }
+                        // As long as the lease is not released, keep renewing the lease in case lease expired since it's still draining. 
+                        if (partition.CurrentOwner != null)
+                        {
+                            this.RenewLease(partition);
+                            renewedLease = true;
+                        }
+                    }
+
+                    //If the lease is other worker's lease. Store it to the dictionary for future balance.
+                    //If the other worker is shutting down, steal the lease.
+                    void CheckOtherWorkerLease(TableLease partition, Dictionary<string, List<TableLease>> partitionDistribution, ReadTableReponse response)
+                    {
+                        bool isOtherWorkerCurrentLease = (partition.CurrentOwner != this.workerName && partition.NextOwner == null && partition.IsDraining == false);
+                        bool isOtherWorkerStolenLease = (partition.CurrentOwner != this.workerName && partition.NextOwner != null);
+                        bool isOwnerShuttingDown = (partition.CurrentOwner != this.workerName && partition.NextOwner == null && partition.IsDraining == true);
+
+                        string owner;
+                        //If the lease is other worker's current lease, add partition to the dictionary with CurrentOwner as key.
+                        if (isOtherWorkerCurrentLease)
+                        {
+                            owner = partition.CurrentOwner!;
+                            this.AddToDictionary(partition, partitionDistribution, owner);
+                        }
+
+                        // If other workers' lease is stolen, suppose lease tranfer could finish successfully, and add partition to the dictionary with NextOwner as key. 
+                        if (isOtherWorkerStolenLease)
+                        {
+                            owner = partition.NextOwner!;
+                            //If the lease was stolen by _this_ worker, we increase its currently owned lease count.
+                            if (owner == this.workerName)
+                            {
+                                ownershipLeaseCount++;
+                                response.WaitForPartition = true;
+                            }
+                            //If the lease is stolen by other workers, add it to the partitionDistribution dictionary with NextOwner as key.
```

Comment:
nit: let's add a space between the "//" and the first character.
Also added a minor phrasing suggestion

```suggestion
                            // If the lease is stolen by another worker, keep track of it for rebalancing purposes
```